apple/validator.go

```diff
@@ -123,7 +123,7 @@ func doRequest(client *http.Client, result interface{}, url string, data url.Val
 	req.Header.Add("accept", AcceptHeader)
 	req.Header.Add("user-agent", UserAgent) // apple requires a user agent
 
-	res, err := client.Do(req)
+	res, err := client.Do(req.WithContext(ctx))
```

Comment:
Instead of changing the context right at the `Do`, can we use `NewRequestWithContext` above so that we are using the context from the start? 

From the [docs](https://golang.org/pkg/net/http/#Request.WithContext) `Between those two uses, it's rare to need WithContext.`Between those two uses, it's rare to need WithContext.`